frontend-ssr/components/tableOfContents/TableOfContents.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import styles from './TableOfContents.module.scss';
+
+interface State {
+    activeTitleChapter: string,
+}
+
+interface Props {
+    chapters: Chapters[],
+}
+
+interface Chapters {
+    title: string,
+    id: string,
+}
+
+class TableOfContents extends React.Component<Props, State> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      activeTitleChapter: null,
+    };
+  }
+
+    handleScrollItem = (id) => {
+      let itemToScroll = document.getElementById(id);
```

Comment:
I just discovered an even easier way to do this, from a friend, including the scroll smooth.  Sorry, I didn't know about this from the beginning. :D 

Instead of using `scrollIntoView` we can change the button to an ```<a href={`#${item.id}`} />```. and then also add in `index.scss` the rule `scroll-behavior: smooth;` on `html` - https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-behavior.

We also need to keep the `onClick` listener but just to set the active item so we know which dot to highlight.

Reply:
Let's also keep the verification regarding `itemToScroll` so we know if there's something missing in the config.